### ISSUE TITLE
Use hugo workflow to publish to github pages

### ISSUE THREAD

**Issue by mattip** [Shekharrajak/poc-hugo#3]
github pages is a nice way to host static content on github itself, with no external hosting needed. This is the way https://numpy.org currently is hosted: merging a PR in the numpy/numpy.org repo automatically regenerates the numpy.org website. [Here](https://gohugo.io/hosting-and-deployment/hosting-on-github/) is a guide to doing this with Hugo.

What do you think?

**Comment by Shekharrajak** [comment]
Yes! I know about it. With surge I want to deploy it in temporary URL for each new PR, so that we can see the changes live. So if PR number 3 is for changing the contents of let say `about.md` ( for example change is : new image is added ) then browsing URL `pr-3-numpyorg.surge.sh`  we should be able to see those changes.

It will help:

1. Easy for review because we don't need screenshot of the change. We can check the change directly online.
2. We can check all styling and UI is working as expected or not.

So final site deployment will remain same .

WDYT?

**Comment by mattip** [comment]
Not sure I understand, maybe it will make more sense when I see the first such URL

**Comment by Shekharrajak** [comment]
You can refer this PR: https://github.com/Shekharrajak/poc-hugo/pull/2

**Comment by Shekharrajak** [comment]
I just create a new blog post: https://github.com/Shekharrajak/poc-hugo/pull/4

**Comment by Shekharrajak** [comment]
> [Here](https://gohugo.io/hosting-and-deployment/hosting-on-github/) is a guide to doing this with Hugo.
> 
> What do you think?

Yes! At the end will be using [Hugo hosting on Github](https://gohugo.io/hosting-and-deployment/hosting-on-github/) for final deployment.
